### ISSUE TITLE
Fetching starred playlist from pyspotify/libspotify failing

### ISSUE THREAD

**Issue by patbenatar** [mopidy/mopidy-spotify#71]
I'm seeing the following error in the mopidy output when trying to get the starred playlist. The trace below shows line 23 in `playlists.py`. I tinkered around a bit and found that when `SpotifyPlaylistsProvider#as_list` is altered to only fetch other (non-starred) playlists via `_get_flattened_playlist_refs()`, it continues to work as expected.

```
ERROR    SpotifyBackend backend caused an exception.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/mopidy/core/playlists.py", line 18, in _backend_error_handling
    yield
  File "/usr/local/lib/python2.7/site-packages/mopidy/core/playlists.py", line 56, in as_list
    playlists = future.get()
  File "/usr/local/lib/python2.7/site-packages/pykka/threading.py", line 52, in get
    compat.reraise(*self._data['exc_info'])
  File "/usr/local/lib/python2.7/site-packages/pykka/compat.py", line 12, in reraise
    exec('raise tp, value, tb')
  File "/usr/local/lib/python2.7/site-packages/pykka/actor.py", line 201, in _actor_loop
    response = self._handle_receive(message)
  File "/usr/local/lib/python2.7/site-packages/pykka/actor.py", line 295, in _handle_receive
    return callee(*message['args'], **message['kwargs'])
  File "/usr/local/lib/python2.7/site-packages/mopidy_spotify/playlists.py", line 23, in as_list
    list(self._get_flattened_playlist_refs()))
  File "/usr/local/lib/python2.7/site-packages/mopidy_spotify/playlists.py", line 37, in _get_starred_playlist_ref

  File "/usr/local/lib/python2.7/site-packages/spotify/playlist.py", line 118, in load
    return utils.load(self._session, self, timeout=timeout)
  File "/usr/local/lib/python2.7/site-packages/spotify/utils.py", line 246, in load
    raise spotify.Timeout(timeout)
Timeout: Operation did not complete in 10.000s
```

If this issue belongs on the `pyspotify` repo, let me know and I'll xpost there.

Thanks!

**Comment by kingosticks** [comment]
Has that traceback been truncated?

**Comment by patbenatar** [comment]
@kingosticks oops, my bad.. updated above.

**Comment by ohjames** [comment]
Spotify servers have been really slow for me lately, I think I've just starred too many songs. Can't use mopidy at the moment due to this error.

**Comment by Rabscuttler** [comment]
I had the same error as @patbenatar, and it was halting spotify integration completely. Deleting line 23 of playlists.py so starred playlists weren't being retrieved did the job.

**Comment by Ticalie** [comment]
@Rabscuttler , Solution confirmed. You should make a pull request :-)

**Comment by jodal** [comment]
While it is okay to remove starred playlists from your own installations to get them working again immediately, simply removing features without understanding why something fails or having a replacement is not okay.

I tested this myself a few days ago. I did not have the described issue, but I noticed that I had two lists named "Starred". One was sorted in the opposite order of the other one. Traditionally, we've made sure to reverse the starred list so that the most recently starred tracks appears at the top, like in the official clients. So it looks like we through libspotify can retrieve the user's starred list like a regular playlist too.

I think the best solution will be to account for this in pyspotify, so the fix benefits all users of pyspotify, not just Mopidy users. I'll have a look at what's needed later this weekend.

**Comment by Rabscuttler** [comment]
@jodal yes indeed - I was not suggesting this as a solution, just a quick fix for a problem that for me completely broke mopidy-spotify.

Thank you for your efforts :+1:

**Comment by kingosticks** [comment]
Like @jodal I don't have the OP's issue but have two starred playlists instead. 

Spotify's 'official' announcement (which they didn't bother forwarding to their developer list for some reason) is [here](https://community.spotify.com/t5/Spotify-Community-Blog/Changes-to-your-Starred-Playlist/ba-p/1202181) and [here](https://support.spotify.com/uk/problems/#!/article/what-happened-to-starred-tracks). Those links would indicate the starred playlist functionality is now redundant and might as well be removed from pyspotify.

**Comment by jodal** [comment]
I'm not very much wiser as to why this works for some and not others. I've cleared out my `~/.cache/mopidy/spotify/` directory and tested a bit. A summary:

- Looking up starred playlists by URIs like `spotify:user:alice:starred` still works
- The starred playlist returned by libspotify still has no name
- Looking up the new "Starred" playlist by the regular playlist URI works
- The new "Starred" playlist has the name "Starred"
- Any track I remove from either of the playlists also disappears from the other
- Any track I do `track.starred = False` on disappears from both playlists

In other words, everything looks like it used to, except I can now see a new "Starred" playlist through the regular playlist API that is kept in sync with my old starred playlist.

**Comment by jodal** [comment]
- If you rename the "Starred" playlist, the sync between the list continues to work
- If you delete the "Starred" playlist from the Spotify client and clears your cache, Mopidy-Spotify can still get hold of both the new "Starred" playlist and the old starred playlist

**Comment by jodal** [comment]
I pushed a fix in bcdcf7afcdcfe6a75f6c6b0f521c64099330dda6. Feel free to test it out and tell me if it fixes the problem for you and I'll roll a new release.

You can install the develop branch by running:

```
pip install https://github.com/mopidy/mopidy-spotify/archive/develop.zip
```

**Comment by gspray** [comment]
This fixed the issue for me. Thanks!

**Comment by jodal** [comment]
Great :-)
